internal-webapps/org.wso2.bfsi.authentication.endpoint/src/main/java/org/wso2/bfsi/authentication/endpoint/BFSIConsentServlet.java

```diff
@@ -0,0 +1,260 @@
+        dispatcher.forward(originalRequest, response);
+
+    }
+
+    /**
+     * Retrieve consent data with the session data key.
+     * @param sessionDataKeyConsent  session data key
+     * @param servletContext         servlet context
+     * @return   HTTP response
+     * @throws IOException if an error occurs while retrieving consent data
+     */
+    HttpResponse getConsentDataWithKey(String sessionDataKeyConsent, ServletContext servletContext) throws IOException {
+
+        String retrievalBaseURL = servletContext.getInitParameter(Constants.RETRIEVAL_BASE_URL);
+        String retrieveUrl = (retrievalBaseURL.endsWith(Constants.SLASH)) ? retrievalBaseURL + sessionDataKeyConsent :
+                retrievalBaseURL + Constants.SLASH + sessionDataKeyConsent;
+
+        CloseableHttpClient client = HttpClientBuilder.create().build();
+        HttpGet dataRequest = new HttpGet(retrieveUrl);
+        dataRequest.addHeader(Constants.AUTHORIZATION, Constants.BASIC + getConsentApiCredentials());
+
+        return client.execute(dataRequest);
+
+    }
+
+    /**
+     * Create consent data from the response of the consent retrieval.
+     * @param consentResponse  consent response from retrieval
+     * @param statusCode       status code of the response
+     * @return  consent data JSON object
+     * @throws IOException if an error occurs while creating the consent data
+     */
+    JSONObject createConsentDataset(JSONObject consentResponse, int statusCode) throws IOException {
+
+        JSONObject errorObject = new JSONObject();
+        if (statusCode != HttpURLConnection.HTTP_OK) {
+            if (statusCode == HttpURLConnection.HTTP_UNAUTHORIZED) {
+                if (consentResponse.has(Constants.DESCRIPTION)) {
+                    errorObject.put(Constants.IS_ERROR, consentResponse.get(Constants.DESCRIPTION));
+                }
+            } else {
+                errorObject.put(Constants.IS_ERROR, "Retrieving consent data failed");
+            }
+            return errorObject;
+        } else {
+            return consentResponse;
+        }
+    }
+
+    /**
+     * Retrieve the config value for Auth servlet Extension.
+     */
+    void setAuthExtension() {
+
+        try {
+            bfsiAuthServletTK = (BFSIAuthServletInterface) Class.forName(parser.getAuthServletExtension())
+                    .getDeclaredConstructor().newInstance();
+        } catch (InstantiationException | IllegalAccessException |
+                InvocationTargetException | NoSuchMethodException | ClassNotFoundException e) {
+            log.error("Webapp extension not found", e);
+        }
+    }
+
+    @Generated(message = "Encapsulated method for unit test")
+    ResourceBundle getResourceBundle(Locale locale) {
+
+        return ResourceBundle.getBundle(BUNDLE, locale);
+    }
+
+    /**
+     * Retrieve admin credentials in Base64 format from webapp properties or OB configs.
+     */
+    static String getConsentApiCredentials () {
```

Comment:
Shouldn't this be an utility method?

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/8/commits/5cab9c9a955547a16e4ab4f4f06ff55db6fb45c5